packages/logger/__tests__/unit/structuredLog/gcpStructuredLog.spec.ts

```diff
@@ -0,0 +1,165 @@
+      options: { includeStackTrace: true },
+      severity: undefined,
+      expected: {
+        message: "test error",
+        severity: "DEFAULT",
+        timestamp: expect.any(String),
+        stackTrace: expect.any(String),
+      },
+    },
+    {
+      id: "4",
+      logData: "another test message",
+      options: { includeStackTrace: false },
+      severity: "info",
+      metadata: {
+        alokai: "test",
+      },
+      expected: {
+        message: "another test message",
+        severity: "INFO",
+        timestamp: expect.any(String),
+        alokai: "test",
+      },
+    },
+    {
+      id: "5",
+      logData: "test message with options",
+      options: { includeStackTrace: false },
+      severity: "debug",
+      expected: {
+        message: "test message with options",
+        severity: "DEBUG",
+        timestamp: expect.any(String),
+      },
+    },
+    {
+      id: "6",
+      logData: new Error("another test error"),
+      options: { includeStackTrace: true },
+      severity: "error",
+      metadata: {
+        alokai: "test",
+      },
+      expected: {
+        message: "another test error",
+        severity: "ERROR",
+        timestamp: expect.any(String),
+        stackTrace: expect.any(String),
+        alokai: "test",
+      },
+    },
+    {
+      id: "7",
+      logData: new Error("another test error"),
+      options: { includeStackTrace: true },
+      severity: "error",
+      metadata: {
+        alokai: "test",
+      },
+      expected: {
+        message: "another test error",
+        severity: "ERROR",
+        timestamp: expect.any(String),
+        stackTrace: expect.any(String),
+        alokai: "test",
+      },
+    },
+    {
+      id: "8",
+      logData: "test message with options",
+      options: { includeStackTrace: false },
+      severity: "debug",
+      expected: {
+        message: "test message with options",
+        severity: "DEBUG",
+        timestamp: expect.any(String),
+      },
+    },
+    {
+      id: "9: with metadata",
+      logData: "message",
+      options: { includeStackTrace: false },
+      severity: "info",
+      metadata: {
+        labels: "other-label",
+      },
+      expected: {
+        message: "message",
+        severity: "INFO",
+        timestamp: expect.any(String),
+        labels: "other-label",
+      },
+    },
+    {
+      id: "10:with ansi codes",
+      logData:
+        "\x1b[90m(/var/www/apps/storefront-unified-nextjs/\x1b[39m.next/server/chunks/689.js:67:9\x1b[90m)\x1b[39m",
+      options: { includeStackTrace: false },
+      severity: "info",
+      metadata: {
+        labels: "other-label",
+      },
+      expected: {
+        message:
+          "(/var/www/apps/storefront-unified-nextjs/.next/server/chunks/689.js:67:9)",
+        severity: "INFO",
+        timestamp: expect.any(String),
+        labels: "other-label",
+      },
+    },
+  ])(
+    "should create a GCP structured log: #ID $id",
```

Comment:
I get the idea of using .each. However, I always find it overwhelming, especially here when we have 10 cases. This test was much harder to grasp than others. Imagine looking at it at midnight when you have to fix smth in a hurry, or just 6 months from now. Please consider refactoring this test to a few granular ones with a detailed description.

Reply:
I think that the matrix for a multiple inputs for the same test is something that we should not fight with. I am very used to such tests. Imagine more input data. This is exactly what it is for and the only argument is that from your POV it is less readable. Please, create a ticket in the jest repo :)